src/libnmstate/schemas/interfaces.json

```diff
@@ -0,0 +1,30 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+
+    "title": "Interfaces tree",
+    "type": "object",
+    "properties": {
+        "interfaces": {
+            "title": "Interfaces",
+            "type": "array",
+            "items": {
+                "title": "Interface",
+                "type": "object",
+                "properties": {
+                    "name": { "type": "string" },
+                    "description": { "type": "string" },
+                    "type": {
+                        "type": "string",
+                        "enum": ["ethernet", "bond", "vlan"]
+                    },
+                    "state": {
+                        "type": "string",
+                        "enum": ["absent", "up", "down", "unknown"]
```

Comment:
is this read only? if not we might want to define a default for it (i don't think it would affect validation, just for documentation purpose, so there is always a state passed to underlying app)

Reply:
The client may not be interested to change the state.